lerp_derive/src/derive.rs

```diff
@@ -53,54 +44,95 @@ impl Parse for LerpAttributes {
     }
 }
 
-fn get_lerp_type(ty: &Type, attrs: &Vec<Attribute>) -> syn::Result<LerpType> {
-    match ty {
-        Type::Path(TypePath { path, .. }) => {
-            let attr = attrs
-                .into_iter()
-                .filter(|attr| attr.path.is_ident("lerp"))
-                .collect::<Vec<_>>();
-
-            let attr: LerpAttributes = match &attr[..] {
-                [] => Ok(Default::default()),
-                [attr] => attr.parse_args(),
-                [_, overflow, ..] => Err(Error::new(
-                    overflow.span(),
-                    "found duplicate attribute on field, consolidate the attributes into one",
-                )),
-            }?;
+fn lerp_field(name: &dyn ToTokens, ty: &Type, attrs: &Vec<Attribute>) -> syn::Result<TokenStream> {
+    let attr = attrs
+        .into_iter()
+        .filter(|attr| attr.path.is_ident("lerp"))
+        .collect::<Vec<_>>();
+
+    let attr: LerpAttributes = match &attr[..] {
+        [] => Ok(Default::default()),
+        [attr] => attr.parse_args(),
+        [_, overflow, ..] => Err(Error::new(
+            overflow.span(),
+            "found duplicate attribute on field, consolidate the attributes into one",
+        )),
+    }?;
 
-            if attr.skip.is_some() {
-                return Ok(LerpType::Skip);
+    match ty {
+        Type::Path(TypePath { path, .. }) => Ok(if attr.skip {
```

Comment:
One final nit, then I think this is done. It looks like we can substantially reduce code duplication by moving the skip check outside the match block, i.e.

```rust
Ok(if attr.skip {quote!{
  #name: self.#name
}} else {
  match ty {
    Type::Path(TypePath { path, .. }) => { ... }
    ...
  }
})
```

Reply:
I have actually just done that in my commit I'm about  to push

Reply:
It also lets you use non-Lerp types in structures and just `#[lerp(skip)]` them